js/ui/windowManager.js

```diff
@@ -1458,6 +1465,13 @@ const WindowManager = new Lang.Class({
                 return;
             let rect = currentSession[0].windowApp.get_frame_rect();
             currentSession[0].buttonApp.set_position(rect.x + rect.width - 100, rect.y + rect.height - 100);
+            if (currentSession[0].windowBuilder === undefined)
+                return;
+            currentSession[0].windowBuilder.move_resize_frame(false,
+                                                              rect.x,
+                                                              rect.y,
+                                                              rect.width,
+                                                              rect.height);
```

Comment:
Does position changed also include size changes?

Also - is there any point to moving around / resizing the (hidden) if it is just going to be move/resized anyway the moment we try and switch to it?